src/preferences/broadcastsettings.cpp

```diff
@@ -1,417 +1,135 @@
-}
-
-QString BroadcastSettings::getDefaultCustomTitle() const {
-    return QString();
-}
-
-bool BroadcastSettings::getEnableMetadata() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kEnableMetadata), getDefaultEnableMetadata());
-}
-
-void BroadcastSettings::setEnableMetadata(bool value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kEnableMetadata), value);
-}
-
-bool BroadcastSettings::getDefaultEnableMetadata() const {
-    return kDefaultEnableMetadata;
-}
-
-bool BroadcastSettings::getEnableReconnect() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kEnableReconnect), getDefaultEnableReconnect());
-}
-
-void BroadcastSettings::setEnableReconnect(bool value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kEnableReconnect), value);
-}
-
-bool BroadcastSettings::getDefaultEnableReconnect() const {
-    return kDefaultEnableReconnect;
-}
-
-// Unused, but we keep this to reserve the name
-bool BroadcastSettings::getEnabled() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kEnabled), true);
-}
-
-void BroadcastSettings::setEnabled(bool value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kEnabled), value);
-}
-
-QString BroadcastSettings::getFormat() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kFormat), getDefaultFormat());
-}
-
-void BroadcastSettings::setFormat(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kFormat), value);
-}
-
-QString BroadcastSettings::getDefaultFormat() const {
-    return QString();
-}
-
-QString BroadcastSettings::getHost() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kHost), getDefaultHost());
-}
-
-void BroadcastSettings::setHost(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kHost), value);
-}
-
-QString BroadcastSettings::getDefaultHost() const {
-    return QString();
-}
-
-bool BroadcastSettings::getLimitReconnects() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kLimitReconnects), getDefaultLimitReconnects());
-}
-
-void BroadcastSettings::setLimitReconnects(bool value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kLimitReconnects), value);
-}
-
-bool BroadcastSettings::getDefaultLimitReconnects() const {
-    return kDefaultLimitReconnects;
-}
-
-QString BroadcastSettings::getLogin() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kLogin), getDefaultLogin());
-}
-
-void BroadcastSettings::setLogin(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kLogin), value);
-}
-
-QString BroadcastSettings::getDefaultLogin() const {
-    return QString();
-}
-
-int BroadcastSettings::getMaximumRetries() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kMaximumRetries), getDefaultMaximumRetries());
+    : m_pConfig(pConfig),
+    m_profiles(),
+    m_currentProfile(kDefaultProfile) {
```

Comment:
Is this prepared for n profiles? If we merge this early, I would like to avoid a future migration for n profiles.